language/diem-framework/compiled/src/shim/tmp_new_transaction_script_builders.rs

```diff
@@ -1438,6 +1465,191 @@ pub enum ScriptCall {
     },
 }
 
+/// Structured representation of a call into a known Move script function.
+/// ```ignore
+/// impl ScriptFunctionCall {
+///     pub fn encode(self) -> TransactionPayload { .. }
+///     pub fn decode(&TransactionPayload) -> Option<ScriptFunctionCall> { .. }
+/// }
+/// ```
+#[derive(Clone, Debug, PartialEq, PartialOrd)]
+#[cfg_attr(feature = "fuzzing", derive(proptest_derive::Arbitrary))]
+#[cfg_attr(feature = "fuzzing", proptest(no_params))]
+pub enum ScriptFunctionCall {
```

Comment:
@matbd @xli @runtian-zhou @bmwill 

Interested to hear your thoughts on the code generation here and how it's structured.

The general methodology is for the legacy transaction scripts, the script builders and types around them remain unchanged (same builder names, same input/output types, same `ScriptCall` enum, same bytecode, everything). Eventually (over the next couple releases) we'll probably want to phase these out.

For script functions, these are represented as a new enum (`ScriptFunctionCall`) and have different transaction builders that return a `TransactionPayload`. All encoders/decoders have the `_function` suffix on them.